src/SpectatorView.Unity/Assets/SpectatorView/Scripts/Networking/INetworkConnection.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Concurrent;
+
+namespace Microsoft.MixedReality.SpectatorView
+{
+    /// <summary>
+    /// Classes that implement this interface represent network connections and allow sending messages across a network.
+    /// </summary>
+    public interface INetworkConnection
+    {
+        /// <summary>
+        /// IP address for the connected peer
+        /// </summary>
+        string Address { get; }
+
+        /// <summary>
+        /// Returns true if the network connection is connected, otherwise false
+        /// </summary>
+        bool IsConnected { get; }
+
+        /// <summary>
+        /// Call to set the network connection state to disconnected
+        /// </summary>
+        void Disconnect();
```

Comment:
Not entirely clear what's the interaction between this and QueueIncomingMessages()

Consider the scenario:

- I received a handshake message.
- I checked some security token.
- It didn't match, so I immediately closed the connection.
- Now I expect that I'll never receive a message from this connection after I closed it, and therefore my message processing code can assume that the token is already verified, and have no checks about that whatsoever.

Do provide a guarantee like this, or some in-flight zombie messages can also be processed afterwards? (the question may be meaningless depending on what QueueIncomingMessages() actually does, please ignore if it doesn't apply)